### PR TITLE
feat: Group call view - Part 1 : creating the grid (AR-1148)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1148" title="AR-1148" target="_blank">StoryAR-1148</a>  View all call participants during the call (grid)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

This PR aims to create the grid that is needed to display the participants tiles in a group call.

- The grid will be created if we have more than 3 participants, otherwise we just display tiles vertically in a column.
- if have more than 8 participants in a call, then we create a new page and we show an indicator to allow user to swipe to next page.
- Tiles sizes are adaptive, they will fit the content space based on how many participants are in the call
- Tiles in the grid contain user info: name, avatar and an indication if he is muted or not.

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Start a group call 🚀

### Notes (Optional)

There is crash happening when the call is established, a fix for that will follow up this PR.

### Attachments (Optional)

https://user-images.githubusercontent.com/18124919/175014090-17ad875b-3da9-4a11-a1e0-5f3fa4b0aca2.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-685/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3693156)(3693156)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3693156)(3693156)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                isMuted = participant.isMuted,
+                isCameraOn = isCameraOn,
+                onSelfUserVideoPreviewCreated = { onSelfVideoPreviewCreated(it) },
+                onClearSelfUserVideoPreview = onSelfClearVideoPreview
+            )
+        }
+    }
+}
+
+private fun tilesRowsCount(value: Int) = if (value % 2 == 0) (value / 2) else ((value / 2) + 1)
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt:61]: Maybe we could explain what exactly this is doing

File: app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        columns = GridCells.Fixed(NUMBER_OF_GRID_CELLS)
+    ) {
+
+        items(participants) { participant ->
+            val numberOfTilesRows = tilesRowsCount(participants.size)
+            val isCameraOn = if (participants.first() == participant) isSelfUserCameraOn else false
+
+            ParticipantTile(
+                modifier = Modifier
+                    .height(((config.screenHeightDp - 178) / numberOfTilesRows).dp)
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/common/CallingGridView.kt:48]: Maybe we could move this into a dimen file

**Review by alexandreferris — APPROVED** [review]
Great work! I just have two small comments in case you have time, otherwise good to go :)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-685/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3693901)(3693901)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3693901)(3693901)-dev-release.apk)

**Pull request merged**